### ISSUE TITLE
Local cache

### ISSUE THREAD

**Issue by multimeric** [vjcitn/rhdf5client#1]
For my use case it's fairly important that I avoid downloading redundant copies of the same HDF5 file. For this reason I wonder if local caching is in scope for this project. For example let's say that the user can configure the cache directory with `options(rhdf5client_cachedir="/tmp/cache")`.

Then, the first time an HDF5 is actually requested, it will get downloaded to that directory and replaced with a normal `HDF5Array`. 

I might be able to find time to implement this myself (not sure), but I wanted to ask here first if this is in scope.

**Comment by multimeric** [comment]
I've had a quick look through the source code to see where this could be added, but it's full of deprecated functions so it's not clear to me which parts are still in use, and where the actual HTTP calls are happening.

**Comment by vjcitn** [comment]
Yes, thanks for keeping this alive.  I've been cleaning up the code base.  Apropos caching, there isn't any download per se.  You issue a query and get back an array of numbers which can be in a binary representation for compactness, or as JSON.  What you do with it is up to you; rhdf5client will produce an R array.

I am going to start a new repo called rhsdsclient where we can explore both the solutions provided in rhdf5client and interfacing to h5pyd, without all the cruft around.  Give me a few minutes.

**Comment by multimeric** [comment]
> Apropos caching, there isn't any download per se. You issue a query and get back an array of numbers which can be in a binary representation for compactness, or as JSON.

Right, but the binary representation could be cached locally, no? For a given query anyway. Also, since you seem to understand this system better than I, how does the HSDS system handle indexing the data to efficiently respond to complex queries?

**Comment by vjcitn** [comment]
Let's use the repo at vjcitn/rhsdsclient for further development.  We can cache locally but at this time this is not part of h5pyd at this time.  There is only one function where httr::GET is used.  I will start an issue on rhsdsclient repo for more details.  Let's move here https://github.com/vjcitn/rhsdsclient/issues/2#issue-1411382548
